hermes/probe/mock/mock_gcs.go

```diff
@@ -0,0 +1,88 @@
+package mock
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/storage"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+)
+
+type fakeBucket struct {
+	attrs   *storage.BucketAttrs
+	objects map[string][]byte
+}
+
+type fakeClient struct {
+	stiface.Client
+	buckets map[string]*fakeBucket
+}
+
+type fakeBucketHandle struct {
+	stiface.BucketHandle
+	c    *fakeClient
+	name string
+}
+
+type fakeObjectHandle struct {
+	stiface.ObjectHandle
+	c          *fakeClient
+	bucketName string
+	name       string
+}
+
+type fakeWriter struct {
+	stiface.Writer
+	obj fakeObjectHandle
+	buf bytes.Buffer
+}
+
+func NewFakeClient() stiface.Client {
+	return &fakeClient{buckets: map[string]*fakeBucket{}}
+}
+
+func (o fakeObjectHandle) NewWriter(context.Context) stiface.Writer {
+	return &fakeWriter{obj: o}
+}
+
+func (b fakeBucketHandle) Create(_ context.Context, _ string, attrs *storage.BucketAttrs) error {
+	if _, ok := b.c.buckets[b.name]; ok {
+		return fmt.Errorf("bucket %q already exists", b.name)
+	}
+	if attrs == nil {
+		attrs = &storage.BucketAttrs{}
+	}
+	attrs.Name = b.name
+	b.c.buckets[b.name] = &fakeBucket{attrs: attrs, objects: map[string][]byte{}}
+	return nil
+}
+
+func (b fakeBucketHandle) Object(name string) stiface.ObjectHandle {
+	return fakeObjectHandle{c: b.c, bucketName: b.name, name: name}
+}
+
+func (c *fakeClient) Bucket(name string) stiface.BucketHandle {
+	return fakeBucketHandle{c: c, name: name}
+}
+
```

Comment:
Add methods to fakeObjectIterator to implement interface.

```suggestion

func (b fakeBucketHandle) Objects(_ context.Context, query *storage.Query) stiface.ObjectIterator {
	iterator := &fakeObjectIterator{
		idx:     0,
		objects: make(map[int]*storage.ObjectAttrs),
	}

	bkt, ok := b.c.buckets[b.name]
	if !ok {
		return nil
	}

	for obj := range bkt.objects {
		if strings.HasPrefix(obj, query.Prefix) {
			iterator.objects[iterator.idx] = &storage.ObjectAttrs{Name: obj}
		}
	}

	return iterator
}

func (i *fakeObjectIterator) Next() (*storage.ObjectAttrs, error) {
	if i.idx >= len(i.objects) {
		return nil, iterator.Done
	}
	obj := i.objects[i.idx]
	i.idx++
	return obj, nil
}

func (i *fakeObjectIterator) PageInfo() *iterator.PageInfo {
	return nil
}

```